src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1148,6 +1164,32 @@ private int GetPromptIndexForPoint(ReadOnlyCollection<SnapshotSpan> sourceSpans,
                 }
                 return index;
             }
+            
+            private bool IsInActivePrompt(SnapshotPoint point)
+            {
+                var editableBuffer = ReadingStandardInput ? StandardInputBuffer : CurrentLanguageBuffer;
+                if (editableBuffer == null)
+                {
+                    return false;
+                }
+                                                                           
+                var sourceSpans = GetSourceSpans(point.Snapshot);
+                var index = GetSourceSpanIndex(sourceSpans, point);
+                if (index == sourceSpans.Count)
+                {
+                    index--;
```

Comment:
That's true in practice, but we probably don't want to depend on it.  I'm very surprised to learn that ``` GetSourceSpanIndex``` behaves that way, but it looks like all callers handle it.